marlowe-playground-client/src/Marlowe/Parser.purs

```diff
@@ -133,6 +247,20 @@ instance valueFromTerm :: FromTerm ValueF where
   fromTerm SlotIntervalEnd = pure SlotIntervalEnd
   fromTerm (UseValue a) = UseValue <$> fromTerm a
 
+instance valueIsMarloweType :: IsMarloweType (ValueF Term) where
+  marloweType _ = ValueType
+
+instance valueHasMarloweHoles :: HasMarloweHoles (ValueF Term) where
+  getHoles m (AvailableMoney a) = getHoles m a
+  getHoles m (Constant a) = insertHole m a
+  getHoles m (NegValue a) = getHoles m a
+  getHoles m (AddValue a b) = getHoles m a <> getHoles m b
+  getHoles m (SubValue a b) = getHoles m a <> getHoles m b
+  getHoles m (ChoiceValue a b) = getHoles m a <> getHoles m b
```

Comment:
I really feel like we're missing a recursion trick here, but it's not a blocker. The lightbulb will go off eventually. :-)

Reply:
indeed but the required monad gets in the way I think, will look again when I get time